Cargill/splinter
# Initial Splinter CLI login subcommand

## Update

To test, using Github oauth

```sh
export OAUTH2_PROVIDER_ID=Github
export OAUTH2_CLIENT_ID="<your Github client id>"
export OAUTH2_CLIENT_SECRET="<your Github client secret>"
export OAUTH2_AUTH_URL="https://github.com/login/oauth/authorize"
export OAUTH2_TOKEN_URL="https://github.com/login/oauth/access_token"
export OAUTH2_SCOPES="user:email"
```

and 

```sh
$ cargo run --manifest-path cli/Cargo.toml --features experimental login
```

You should have a file at `$HOME/.splinter/auth_token` which contains the token values from the provider.

For Google,

```sh
export OAUTH2_PROVIDER_ID=Google
export OAUTH2_CLIENT_ID="<your Google client id>"
export OAUTH2_CLIENT_SECRET="<your Google client secret>"
export OAUTH2_AUTH_URL="https://accounts.google.com/o/oauth2/v2/auth"
export OAUTH2_TOKEN_URL="https://oauth2.googleapis.com/token"
export OAUTH2_SCOPES="openid,email,profile"
```

(This probably requires a clean build between changes to these env variables.)

## Note

The compile-time configuration will be replaced with a call to a Splinter REST API endpoint, such that the variables match.
